src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -918,5 +918,92 @@ public static void ConverterRead_VerifyInvalidTypeToConvertFails()
                 }
             }
         }
+
+        [Theory]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(List<int>))]
+        [InlineData(typeof(Dictionary<int, string>))]
+        public static void GetTypeInfo_MutableOptionsInstance(Type type)
+        {
+            var options = new JsonSerializerOptions();
+            JsonTypeInfo typeInfo = options.GetTypeInfo(type);
```

Comment:
honestly I think it would be ok if we locked options when it's called

Reply:
After further investigations, I have concluded that generating metadata with mutable options does not actually compromise the cache. Plus, this makes semantics consistent with how `GetConverter` behaves.